src/style.rs

```diff
@@ -631,4 +826,67 @@ mod tests {
         // reset
         assert_eq!(Style::new().reset(), Style::reset());
     }
+
+    #[test]
+    #[cfg(feature = "crossterm")]
+    fn from_crossterm_content_style() {
+        use crossterm::style::{
+            Attribute as CAttribute, Attributes as CAttributes, Color as CColor, ContentStyle,
+        };
+
+        let content_styles = [
+            ContentStyle::default(),
+            ContentStyle {
+                foreground_color: Some(CColor::DarkYellow),
+                ..Default::default()
+            },
+            ContentStyle {
+                background_color: Some(CColor::DarkYellow),
+                ..Default::default()
+            },
+            ContentStyle {
+                attributes: CAttributes::from(CAttribute::Bold),
+                ..Default::default()
+            },
+            ContentStyle {
+                attributes: CAttributes::from(CAttribute::NoBold),
+                ..Default::default()
+            },
+            ContentStyle {
+                attributes: CAttributes::from(CAttribute::Italic),
+                ..Default::default()
+            },
+            ContentStyle {
+                attributes: CAttributes::from(CAttribute::NoItalic),
+                ..Default::default()
+            },
+            ContentStyle {
+                attributes: CAttributes::from([CAttribute::Bold, CAttribute::Italic].as_ref()),
+                ..Default::default()
+            },
+            ContentStyle {
+                attributes: CAttributes::from([CAttribute::NoBold, CAttribute::NoItalic].as_ref()),
+                ..Default::default()
+            },
+        ];
+        let styles = styles();
+
+        for (content_style, style) in content_styles.into_iter().zip(styles) {
+            assert_eq!(Style::from(content_style), style);
+        }
```

Comment:
When reading it I had to read down, then up to work out what fn styles was doing and then for each style had to read up and down to keep them in sync. This is a test smell. I'd prefer to replace this approach with many asserts like:
```rust
assert_eq!(ContentStyle::new().dark_yellow().into(), Style::new().yellow());
```

Reply:
You're right I also had to do that while writing the test. Fixed when moved to crossterm